### PR TITLE
Migrate to Kotlin DSL

### PR DESCRIPTION
**Pull request by johnshea**

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            )
+        }
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
+    }
+    kotlinOptions {
+        jvmTarget = "17"
+        freeCompilerArgs += "-opt-in=androidx.compose.material3.ExperimentalMaterial3Api"
```
Comment (jhansitavva) [inline app/build.gradle.kts]: Remove `freeCompilerArgs`.

File: app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        freeCompilerArgs += "-opt-in=androidx.compose.material.ExperimentalMaterialApi"
+    }
+    buildFeatures {
+        viewBinding = true
+        // Enables Jetpack Compose for this module
+        compose = true
+    }
+    composeOptions {
+        kotlinCompilerExtensionVersion = "1.4.7"
```
Comment (jhansitavva) [inline app/build.gradle.kts]: As mentioned in a different PR, let's be consistent unless this matches with the starter code.

Comment (jhansitavva) [reply app/build.gradle.kts]: Please ignore if this is done on purpose to resolve codelab issues.

**Review by jhansitavva — COMMENTED** [review]
Same feedback as https://github.com/google-developer-training/basic-android-kotlin-compose-training-juice-tracker/pull/17 
Added couple more!

File: app/build.gradle.kts (hunk 1 above)
Comment (johnshea) [reply app/build.gradle.kts]: Removed.

File: app/build.gradle.kts (hunk 2 above)
Comment (johnshea) [reply app/build.gradle.kts]: Ack. No change as this is added this way per the codelab instructions. Does not exist in starter code.

**Comment by johnshea** [comment]
Upgraded to latest versions similar to PR https://github.com/google-developer-training/basic-android-kotlin-compose-training-juice-tracker/pull/17

**Review by jhansitavva — APPROVED** [review]

**Pull request merged**